MapleServer2/Data/Static/ItemStatsMetadataStorage.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using Maple2Storage.Types.Metadata;
+using MapleServer2.Constants;
+using ProtoBuf;
+
+namespace MapleServer2.Data.Static
+{
+    public static class ItemStatsMetadataStorage
+    {
+        private static readonly Dictionary<int, List<ItemOptions>> Constant = new Dictionary<int, List<ItemOptions>>();
+        private static readonly Dictionary<int, List<ItemOptions>> Random = new Dictionary<int, List<ItemOptions>>();
+        private static readonly Dictionary<int, List<ItemOptions>> Static = new Dictionary<int, List<ItemOptions>>();
+
+        static ItemStatsMetadataStorage()
+        {
+            using FileStream stream = File.OpenRead($"{Paths.RESOURCES}/ms2-item-stats-metadata");
+            List<ItemStatsMetadata> items = Serializer.Deserialize<List<ItemStatsMetadata>>(stream);
+            foreach (ItemStatsMetadata item in items)
+            {
+                if (!Constant.ContainsKey(item.ItemId))
+                {
+                    Constant[item.ItemId] = item.Constant;
+                }
+                else
+                {
+                    Constant[item.ItemId].AddRange(item.Constant);
+                }
+                if (!Random.ContainsKey(item.ItemId))
+                {
+                    Random[item.ItemId] = item.Random;
+                }
+                else
+                {
+                    Random[item.ItemId].AddRange(item.Random);
+                }
+                if (!Static.ContainsKey(item.ItemId))
+                {
+                    Static[item.ItemId] = item.Static;
+                }
+                else
+                {
+                    Static[item.ItemId].AddRange(item.Static);
+                }
+            }
+
+        }
+
+        public static bool HasConstantStats(int itemId) => Constant.ContainsKey(itemId);
+
+        public static bool HasRandomStats(int itemId) => Random.ContainsKey(itemId);
+
+        public static bool HasStaticStats(int itemId) => Static.ContainsKey(itemId);
+
+        public static bool GetConstantStat(int itemId, out List<ItemOptions> list)
+        {
+            list = null;
```

Comment:
Doesn't it already default to null?

Reply:
It doesn't. "the parameter out "list" must be assigned before the control leaves the current method". 
I think another way is set list to null before the `return false`;

Reply:
Just curious, what about assigning it null in the parameter?

Reply:
Can't also. 

`CS1741 | Error | A ref or out parameter cannot have a default value`

Reply:
Alright sounds good